File: StructuralPatterns/Adapter/TypeScript/Adapter.ts
import { Target } from './Target';
import { Adaptee } from './Adaptee';

export class Adapter implements Target {
    private adaptee: Adaptee;

    constructor(adaptee: Adaptee) {
        this.adaptee = adaptee;
    }

    request(): string {
        return this.adaptee.specificRequest().split('').reverse().join('');
    }
}

File: StructuralPatterns/Adapter/CSharp/Adapter.cs
using System;

namespace Adapter;

public class Adapter : ITarget
{
    private readonly Adaptee _adaptee;

    public Adapter(Adaptee adaptee)
    {
        _adaptee = adaptee;
    }

    public string Request()
    {
        var charArray = _adaptee.SpecificRequest().ToCharArray();
        Array.Reverse(charArray);
        return new string(charArray);
    }
}
